### DISCUSSION TITLE
SpellChecker not built

### DISCUSSION THREAD

**Discussion by sodiumchl** [ONLYOFFICE/build_tools#433] [Q&A] [answered]
After running `tools/linux/automate.py server`, SpellChecker is not present in the out folder.
It seems the following lines are needed in scripts/build_server.py: (Or maybe I'm not doing something correctly?)

  base.copy_file(server_dir + '/SpellChecker/package.json', server_build_dir + '/SpellChecker')
  base.copy_file(server_dir + '/SpellChecker/npm-shrinkwrap.json', server_build_dir + '/SpellChecker')
  base.cmd_in_dir(server_build_dir + "/SpellChecker", "npm", ["install"])
  base.cmd_in_dir(server_build_dir + "/SpellChecker", "pkg", [".", "-t", pkg_target, "-o", "spellchecker"])

**Comment by sodiumchl** [comment]
Also, the following lines are needed in nginx config:

upstream spellchecker {
  server localhost:8080 max_fails=0 fail_timeout=0s;
}

**Comment by ShockwaveNN** [comment, marked as answer]
Since [v6.3.0](https://github.com/ONLYOFFICE/DocumentServer/blob/master/CHANGELOG.md#all-editors-5) Spellchecker is not a separate server, so no need to run it at all (although some leftovers may still in our code, but separate service is not required)

**Comment by sodiumchl** [comment]
I see. Thank you!
